rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSourceDescriptor.java

```diff
@@ -0,0 +1,198 @@
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.resourcemanager.nodesource;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.ow2.proactive.resourcemanager.authentication.Client;
+
+
+/**
+ *
+ */
+public class NodeSourceDescriptor implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private String name;
+
+    private String infrastructureType;
+
+    private Object[] infrastructureParameters;
+
+    private String policyType;
+
+    private Object[] policyParameters;
+
+    private Client provider;
+
+    private boolean nodesRecoverable;
+
+    private NodeSourceStatus status;
+
+    private Map<String, Serializable> lastRecoveredInfrastructureVariables;
+
+    private NodeSourceDescriptor(String name, String infrastructureType, Object[] infrastructureParameters,
+            String policyType, Object[] policyParameters, Client provider, boolean nodesRecoverable,
+            NodeSourceStatus status) {
+        this.name = name;
+        this.infrastructureType = infrastructureType;
+        this.infrastructureParameters = infrastructureParameters;
+        this.policyType = policyType;
+        this.policyParameters = policyParameters;
+        this.provider = provider;
+        this.nodesRecoverable = nodesRecoverable;
+        this.status = status;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public String getInfrastructureType() {
+        return this.infrastructureType;
+    }
+
+    public Object[] getInfrastructureParameters() {
+        return this.infrastructureParameters;
+    }
+
+    public String getPolicyType() {
+        return this.policyType;
+    }
+
+    public Object[] getPolicyParameters() {
+        return this.policyParameters;
+    }
+
+    public Client getProvider() {
+        return this.provider;
+    }
+
+    public boolean nodesRecoverable() {
+        return this.nodesRecoverable;
+    }
+
+    public NodeSourceStatus getStatus() {
+        return this.status;
+    }
+
+    public Map<String, Serializable> getLastRecoveredInfrastructureVariables() {
+        return this.lastRecoveredInfrastructureVariables;
+    }
+
+    public void setStatus(NodeSourceStatus status) {
+        this.status = status;
+    }
+
+    private void
+            setLastRecoveredInfrastructureVariables(Map<String, Serializable> lastRecoveredInfrastructureVariables) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this method into "Builder". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3398)